### ISSUE TITLE
why can not reach high hit@10

### ISSUE THREAD

**Issue by qqt-code** [HKUST-KnowComp/UC-FedRec#1]
i have runned main_FedRec_ML.py，and the hype paramaters in this experiment are below：
![image](https://github.com/HKUST-KnowComp/UC-FedRec/assets/63205718/4ee5447e-fd22-4458-b03a-ee32240033c4)
but even if in 679 epoch，the hit@10 of result only reach 0.22,but in the paper, it can reach about 0.8,i want to know if i have some ploblems in running this python file.
![image](https://github.com/HKUST-KnowComp/UC-FedRec/assets/63205718/d26afbe5-274e-45b2-a635-f847cd1f335b)

**Comment by liiixiiiin** [comment]
> i have runned main_FedRec_ML.py，and the hype paramaters in this experiment are below： ![image](https://private-user-images.githubusercontent.com/63205718/339665380-4ee5447e-fd22-4458-b03a-ee32240033c4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjgxOTUyODgsIm5iZiI6MTcyODE5NDk4OCwicGF0aCI6Ii82MzIwNTcxOC8zMzk2NjUzODAtNGVlNTQ0N2UtZmQyMi00NDU4LWIwM2EtZWUzMjI0MDAzM2M0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDA2VDA2MDk0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFmMjFmODBjZjc4NmE2MTM5NjU5MDhjNWQ2ZjgxODFiNDI0M2I5OTFjOWMxMWY4Y2Y3MzEwZWZjMjFkYTYwNjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.AXXp9PPIeqzsXpAk3FHPJpLPf_CbcV3anTomEHOnY-U) but even if in 679 epoch，the hit@10 of result only reach 0.22,but in the paper, it can reach about 0.8,i want to know if i have some ploblems in running this python file. ![image](https://private-user-images.githubusercontent.com/63205718/339666095-d26afbe5-274e-45b2-a635-f847cd1f335b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjgxOTUyODgsIm5iZiI6MTcyODE5NDk4OCwicGF0aCI6Ii82MzIwNTcxOC8zMzk2NjYwOTUtZDI2YWZiZTUtMjc0ZS00NWIyLWE2MzUtZjg0N2NkMWYzMzViLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDA2VDA2MDk0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU4OTQ0NzQ3NDllNDIwZDg3OWU0M2QyY2ExMGM2NDRhZjc3NjczZDM2N2I5NGRmMzY5MDJjNmY0YTk3YWUyMjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Ll6tZ_M_vaBSg8LB6bkTERHK-acCyi7XNawVeYJ96eM)

I got the same result as you , can you imporve it right now?

**Comment by strygwyr76** [comment]
To be honest I ran this experiment 3 years ago and cannot remember some details. The paper takes a long time to be accepted. But I will try my best to give you suggestions. The model performance is too low, indicating that it does not learn anything.

As shown in your figure, the model achieves the best performance in the 196 epoch, much earlier than I did. I remember that the model converges at around 2000 epochs, maybe you can extend the training time. 

Besides, the model is sensitive to hyperparameters, you can tune some hyperparameters to see whether the performance can be improved.

Although I cannot remember all the details, I found a log file on the server.  You can check the file to see whether it helps. [https://github.com/HKUST-KnowComp/UC-FedRec/blob/main/Dist_no_pri_run4_1.log](url)

If you have any questions, you can contact me through email, the GitHub issue does not send me notifications.

**Comment by strygwyr76** [comment]
[Uploading Dist_no_pri_run4_1.log…]()

**Comment by qqt-code** [comment]
Just need to increase the learning rate; I vaguely remember it should be increased tenfold from the default value. You can try it out more.
